src/adapter/ads/nativeads.js

```diff
@@ -49,30 +50,101 @@ let NativeAdsAdapter = youbora.StandardAdapter.extend({
     return returnValue;
   },
 
-  /**  @returns {void} - Return a list of events and methods to suscribe from this.player. */
-  getListenersList: function() {
+  /** @returns {object} - given ad structure (list with number of pre, mid, and post breaks) */
+  getGivenBreaks: function() {
+    return this.adBreaks ? this.adBreaks.length : null;
+  },
+
+  /** @returns {Array} - a list of playheads of ad breaks begin time */
+  getBreaksTime: function() {
+    let ret = this.adBreaks;
+    if (ret) {
+      let len = ret.length;
+      if (len > 0 && ret[len - 1] === -1) {
+        ret[len - 1] = this.plugin.getDuration() || -1;
+      }
+    }
+    return ret;
```

Comment:
it seems you are running over the plugin's adBreaks object so it will be changed once this method is run.
Is this expected?

Reply:
Yes, we are going to call methods using adBreaks object only at the beginning of the views